simulator/__init__.py

```diff
@@ -198,15 +205,15 @@ def train_population_from_file_random_program(individuals_path, programs_dict, t
     individuals = []
     for _, individual_series in individuals_df.iterrows():
         individuals.append(Individual(series=individual_series))
-
+    for individual in individuals:
         # Get a random program
         training_program_path = programs_dict[random.randint(
             1, len(programs_dict))]
-    timestamp = __train_and_save(
-        individuals, training_results_path, training_program_path)
+        timestamp = __train_and_save(
+        [individual], training_results_path, training_program_path)
 
-    # add the updated individuals to bottom of csv file
-    save_individuals(individuals, individuals_path, timestamp)
+        # add the updated individuals to bottom of csv file
+        save_individuals([individual], individuals_path, timestamp)
```

Comment:
Could we perhaps iterate through the for-loop and append individual, individuals_path and timestamp to a list (if necessary for each individual) and then save them all at once? Don't know if you're having performance issues but that could speed it up quite a bit if that is the case. Saving a lot of data on slow memory should be a lot faster than saving it individually

Reply:
Yeah makes sense to do that from a time-complexity perspective, but I think it might use too much RAM then?